Tests/MongoSwiftTests/CommandMonitoringTests.swift

```diff
@@ -84,7 +84,7 @@ final class CommandMonitoringTests: MongoSwiftTestCase {
 
     func testAlternateNotificationCenters() throws {
         let client = try MongoClient(options: ClientOptions(eventMonitoring: true))
-        let db = try client.db("commandTest")
+        let db = try client.db(type(of: self).testDatabase)
```

Comment:
Why is `type(of:)` required? Wouldn't this be similar to how we call `self.getSpecsPath()` in other tests, even though that method is defined on XCTestCase (via an extension)?

Reply:
I don't believe we defined such a method. We do have the static computed property `specsPath` on `MongoSwiftTestCase`, which we refer to by `MongoSwiftTestCase.specsPath` in all our tests. 

Using `type(of:)` to access `testDatabase` is slightly less brittle in the event we rename `MongoSwiftTestCase`, and, more importantly, it allows us to override `testDatabase` in subclasses without updating how we refer to it.

Reply:
Oops, I was looking at an outdated git branch when I mentioned `getSpecsPath()`. It has since been replaced with a static var on MongoSwiftTestCase.

Based on [this article](https://www.hackingwithswift.com/example-code/language/whats-the-difference-between-a-static-variable-and-a-class-variable), my understanding is that `static` and `class` only differ in whether they allow inheritance (unless that's an over simplification). I would have expected that `self` here would refer to the compile-time class where this method is defined (i.e. CommandMonitoringTests). So if CommandMonitoringTests had overwritten `testDatabase`, that new value would be returned. I would think that `type(of: self)` is more useful in a base class (e.g. a test helper method on MongoSwiftTestCase). From my experience with PHP, that would be a form of [late static binding](http://php.net/manual/en/language.oop5.late-static-bindings.php).

I admittedly haven't worked with this in Swift, so I'm trying to wrap my head around it. Happy to defer to the other reviewers that this is correct, though.